Implement WriteBuffer float, double and BigInteger functions

hutcheb opened the pull request:
Initial commit, hasn't been tested at all.

hutcheb committed:
Draft for writeFloat in WriteBUffer

hutcheb committed:
Add support for writing floating point.

Note knx driver uses a 16-bit floating point. The logic needs to be 
reviewed and tested.

chrisdutz commented on plc4j/drivers/knxnetip/src/main/java/org/apache/plc4x/java/knxnetip/utils/KnxHelper.java:
```diff
@@ -46,9 +46,16 @@ public static void f16toBytes(WriteBuffer io, Object param) {
+            //io.writeInt(4, exponent);
+            //io.writeDouble(11, Math.getExponent(mantissa), mantissa); //Don't think this works BH
+
+            String mantissaString = Double.toString(mantissa);
+            int mantissaInteger = Integer.parseInt(mantissaString.substring(mantissaString.indexOf('.')));
```
What's the String conversion for? I think in this case a simple double-to-int-cast should do the trick ....

hutcheb replied:
The string conversion is to get the digits after the decimal place of the mantissa as part of writing a 16-bit float.
(In my head the mantissa is always going to have a one before the decimal place which can be discarded)
The digits after are then converted to an integer to be written as the manitssa part.

I haven't tested this part of it, it has a slight chance of working for most cases but not all. I'll create a 16-bit floating point PlcValue at some point and test it with Modbus and then move it over to the KNX protocol as a custom PlcValue.

chrisdutz reviewed (approved):
LGTM :-) Thanks

hutcheb committed:
Add support for Big Interger write (untested)

hutcheb committed:
replace Modbus fromPLCValue function with staticSerialise functions. 
(Needs to be fixed to include changes from Data Type PR)

hutcheb committed:
Merge branch 'develop' into Bug/AddWriteBufferSupportFloatBigInt

hutcheb committed:
Moved Modbus over to use the static Serializer.
I'll do some more testing tomorrow, but seems to work with Big Ints and 
Floats

hutcheb committed:
Fixed support for ULINT writing

hutcheb committed:
Refactored the BigInteger functions

hutcheb commented:
Little Endian BigInteger functions havn't been tested.

hutcheb committed:
Implemented staticSerialize write list functions

Outstanding:-
- Check size of buffer when writing < 8 bits
- Find out why writing < 8 bits results in a byte of all zeros.
- check endianness when writing bits.
